gcn-alert/gcn_kafka_receiver.py

```diff
@@ -56,12 +67,15 @@ def post_to_slack(channel, message):
         try:
             alert = json.loads(message.value().decode('utf-8'))
             
-            rate_duration = alert["rate_duration"]
-            event_time = datetime.strptime(alert["trigger_time"], '%Y-%m-%dT%H:%M:%S.%fZ')
+            rate_duration = alert.get("rate_duration", None)
+            event_time_str = alert.get("trigger_time", None)
+            event_time = datetime.strptime(event_time_str, '%Y-%m-%dT%H:%M:%S.%fZ') if event_time_str else None
             current_time = datetime.utcnow()
 
             if (
-                rate_duration < 2 
+                rate_duration is not None 
+                and event_time is not None
+                and rate_duration < 2
```

Comment:
if `rate_duration` is `None`, then I think this will produce a `TypeError`. Need to test for `None` before testing if value is less than 2.